core/container.go

```diff
@@ -391,7 +393,7 @@ func (container *Container) Build(
 ) (*Container, error) {
 	container = container.Clone()
 
-	container.Services.Merge(contextDir.Services)
+	container.Services.Merge(dockerfileDir.Services)
```

Comment:
I think we probably need to include the `contextDir` services as well?

Reply:
The services map is used to spin up any dependent services right before we actually execute an operation that might require the use of them. We need to track them correctly, or we can fail to evaluate certain things.

It shouldn't affect things for this code here, but if we ever change the calling code where `contextDir` isn't a subset, then we'd hit this codepath.